src/utilities/validator.js

```diff
@@ -37,17 +32,26 @@ const schema = joi.object({
     // object containing a string path and React component
     joi.object({
       component: joi.func(),
+      endpoint: joi.alternatives().try(joi.string(), joi.func()),
       getComponent: joi.func(),
-      path: joi.string(),
-      endpoint: joi.alternatives().try(joi.string(), joi.func())
+      path: joi.string()
     })
   ),
-  // optional array of proxy paths
+  // optional array of proxy paths e.g. ['path', 'another/path']
   proxyPaths: joi.array().items(joi.string()),
-  // optional object of redirects
-  redirectPaths: joi.object(),
+  // optional object of redirects e.g. { 'path': 'to-redirect' }
+  redirectPaths: joi.object().pattern(/.*/, joi.string()),
   // optional function run when routing on the client
-  onPageUpdate: joi.func()
+  onPageUpdate: joi.func(),
+  // misc options
+  options: joi.object().keys({
+    // string e.g. 'localhost', '0.0.0.0'
+    host: joi.string(),
+    // number e.g. 3030
+    port: joi.number(),
+    // theme color for progress bar
+    themeColor: joi.string().hex()
```

Comment:
Not currently, I'm thinking a bit fewtch here, I guess it's unlikely... I'll switch it to `progressBarColor`